certbot-dns-linode/certbot_dns_linode/dns_linode.py

```diff
@@ -52,19 +56,44 @@ def _cleanup(self, domain, validation_name, validation):
         self._get_linode_client().del_txt_record(domain, validation_name, validation)
 
     def _get_linode_client(self):
-        return _LinodeLexiconClient(self.credentials.conf('key'))
+        api_key = self.credentials.conf('key')
+        api_version = self.credentials.conf('version')
+
+        if not api_version:
+            api_version = 3
+
+            # Match for v4 api key
+            regex_v4 = re.compile('^[0-9a-f]{64}$')
+            regex_match = regex_v4.match(api_key)
+            if regex_match:
+                api_version = 4
+        else:
+            api_version = int(api_version)
+
+        return _LinodeLexiconClient(api_key, api_version)
 
 class _LinodeLexiconClient(dns_common_lexicon.LexiconClient):
     """
     Encapsulates all communication with the Linode API.
     """
 
-    def __init__(self, api_key):
+    def __init__(self, api_key, api_version):
         super(_LinodeLexiconClient, self).__init__()
-        self.provider = linode.Provider({
-            'provider_name': 'linode',
-            'auth_token': api_key
-        })
+        self.api_version = api_version
+
+        if api_version == 3:
+            self.provider = linode.Provider({
+                'provider_name': 'linode',
+                'auth_token': api_key
+            })
+        elif api_version == 4:
+            self.provider = linode4.Provider({
+                'provider_name': 'linode4',
+                'auth_token': api_key
+            })
+        else:
+            raise errors.PluginError('Invalid api version specified: {0}.'
```

Comment:
To make this a bit more future-proof (as Linode may eventually introduce a v5, at which point this error may be confusing), it might be helpful to list the supported API versions:
```suggestion
             raise errors.PluginError('Invalid api version specified: {0}. (Supported: 3, 4.)'
```

Reply:
I think when v5 does come around, v3 will be long gone but I'll make the changes.